### PR TITLE
PR4: Export settings support added (Rushikesh)

### PR DESCRIPTION
**Pull request by ruuushhh**

### CODE REVIEW

File: apps/workspaces/serializers.py [hunk 1]
```diff
@@ -76,4 +78,38 @@ class Sage300CredentialSerializer(serializers.ModelSerializer):
...
+        read_only_fields = ('id', 'workspace', 'created_at', 'updated_at')
+
+    def create(self, validated_data):
+        """
+        Create Export Settings
+        """
+        assert_valid(validated_data, 'Body cannot be null')
+        workspace_id = self.context['request'].parser_context.get('kwargs').get('workspace_id')
+
+        export_settings = ExportSettings.objects.filter(
```
Comment (NileshPant1999) [inline apps/workspaces/serializers.py:100]: why are we doing this ??

File: apps/workspaces/migrations/0004_auto_20231017_1843.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+# Generated by Django 3.2.22 on 2023-10-17 18:43
+
+from django.db import migrations, models
+import django.db.models.deletion
+import sage_desktop_api.helpers
+
+
+class Migration(migrations.Migration):
```
Comment (NileshPant1999) [inline apps/workspaces/migrations/0004_auto_20231017_1843.py:8]: @ruuushhh please make sure to rerun the migration after other branches are merged to master

File: apps/workspaces/views.py [hunk 3]
```diff
@@ -39,7 +38,7 @@ class WorkspaceView(generics.CreateAPIView, generics.RetrieveAPIView):
     """
     serializer_class = WorkspaceSerializer
 
-    permission_classes = [
+    permission_classes = [  # Remove IsAuthenticated once we have a proper auth flow
```
Comment (NileshPant1999) [inline apps/workspaces/views.py:41]: @ruuushhh we can remove this, this will be fixed once we uncomment default permission class from settings file

File: sage_desktop_api/tests/settings.py [hunk 4]
```diff
@@ -162,7 +162,13 @@
     }
 }
 
-DATABASE_ROUTERS = ['sage_desktop_api.cache_router.CacheRouter']
+# DATABASES['cache_db'] = {
+#     'ENGINE': 'django.db.backends.sqlite3',
```
Comment (NileshPant1999) [inline sage_desktop_api/tests/settings.py:166]: can delete this too, no need to comment

File: sage_desktop_api/models/fields.py [hunk 5]
```diff
@@ -29,54 +54,41 @@ class TextNotNullField(models.TextField):
...
-        kwargs['null'] = False  # Ensure the field is not nullable
+        kwargs['null'] = False
         kwargs['help_text'] = kwargs.get('help_text', 'text field with null false')
         super(TextNotNullField, self).__init__(*args, **kwargs)
 
 class StringOptionsField(models.CharField):
     description = "Custom String Field with Options"
 
     def __init__(self, *args, **kwargs):
-        choices = kwargs.pop('choices', [])  # Retrieve choices from kwargs
```
Comment (NileshPant1999) [inline sage_desktop_api/models/fields.py:40]: why did you delete these comments ??

File: sage_desktop_api/tests/settings.py (hunk 4 above)
Comment (NileshPant1999) [reply sage_desktop_api/tests/settings.py:162]: @ruuushhh you marked it as resolved, but did not deleted it ??

File: apps/workspaces/models.py [hunk 6]
```diff
@@ -107,9 +195,9 @@ class AdvancedSetting(BaseModel):
     schedule_is_enabled = BooleanFalseField(help_text='Boolean to check if schedule is enabled')
     schedule_start_datetime = CustomDateTimeField(help_text='Schedule start date and time')
     schedule_id = StringNullField(help_text='Schedule id')
-    interval_hours = models.IntegerField(null=True)
-    emails_selected = models.JSONField(default=list, help_text='Emails Selected For Email Notification',  null=True)
-    emails_added = models.JSONField(default=list, help_text='Emails Selected For Email Notification',  null=True)
+    interval_hours = IntegerNullField(help_text='Interval in hours')
+    emails_selected = CustomeJsonField(help_text='Emails Selected For Email Notification')
```
Comment (Shwetabhk) [inline apps/workspaces/models.py:199]:
```suggestion
    emails_selected = CustomJsonField(help_text='Emails Selected For Email Notification')
```

**Review by Shwetabhk — APPROVED** [review]
